File: forntend_quanlynhansu/src/services/WorkScheduleService.js
import api from "./BaseApi";

const WorkScheduleService = {
    getAll: () => api.get("/schedules"),
    getById: (id) => api.get(`/schedules/${id}`),
    create: (data) => api.post("/schedules", data),
    update: (id, data) => api.put(`/schedules/${id}`, data),
    delete: (id) => api.delete(`/schedules/${id}`),
    getByUserId: (userId) => api.get(`/schedules/user/${userId}`),
    createGroup: (data) => api.post("/schedules/group", data)
};

export default WorkScheduleService;

File: src/main/java/com/example/quanlynhansu/service/WorkScheduleService.java
package com.example.quanlynhansu.service;

import com.example.quanlynhansu.dto.WorkScheduleRequest;
import com.example.quanlynhansu.entity.WorkSchedule;

import java.util.List;

public interface WorkScheduleService {
    WorkSchedule create(WorkSchedule schedule);
    WorkSchedule update(Long id, WorkSchedule schedule);
    void delete(Long id);
    WorkSchedule getById(Long id);
    List<WorkSchedule> getAll();
    List<WorkSchedule> getSchedulesByUserId(Long userId);
    List<WorkSchedule> createMany(WorkScheduleRequest request);
}
